pkgs/top-level/all-packages.nix

```diff
@@ -28729,13 +28729,13 @@ with pkgs;
   zfs_2_1 = callPackage ../os-specific/linux/zfs/2_1.nix {
     configFile = "user";
   };
-  zfsStable = callPackage ../os-specific/linux/zfs/stable.nix {
+  zfs_2_2 = callPackage ../os-specific/linux/zfs/2_2.nix {
     configFile = "user";
   };
-  zfsUnstable = callPackage ../os-specific/linux/zfs/unstable.nix {
+  zfs_unstable = callPackage ../os-specific/linux/zfs/unstable.nix {
     configFile = "user";
   };
-  zfs = zfsStable;
+  zfs = zfs_2_2;
```

Comment:
This is the one annoying unanticipated edge case with RFC 140. I do like how it encourages refactorings though: In this case a `pkgs.zfsVersions` like described in https://github.com/NixOS/nixpkgs/pull/287918#issuecomment-1939243084 would work great, I'd love to see that, it's my recommendation here.

However, merging this PR as is indeed won't break master (I have [a PR](https://github.com/NixOS/nixpkgs/pull/290743) to make this clearer), but we really shouldn't normalise merging with red checks (that's how we ended up with https://github.com/NixOS/nixpkgs/issues/289649).

Instead I'd like to improve the error messages and documentation to show how such refactorings can be done (and manually responding to pings as here until then).

---

As a last resort, it's also possible to weaken the check for new packages to not trigger in such cases. This will be easy to do once the code for the automated migration is in place (working on it). I hope this won't be necessary with the improved error message/docs though.

Reply:
> Package sets also have downsides: they are more awkward to overlay and adding aliases requires foregoing all the tooling builtin to `aliases.nix` and manually dealing with `allowAliases`.

Oh yeah that's a good point. I just opened https://github.com/NixOS/nixpkgs/pull/292214 to have a better recommendation that can be applied more generally.

> One alternative that to consider (nowhere near an opinion) is changing pkgs/by-name migration check to add a comment to the PR when merging is still “okay but discouraged”, rather than a failing check.

That's a good idea, I'll keep it in mind and might use it if this becomes a bigger problem. For now I'm hoping the PR I just opened will give a good path forward, and once merged I intend to change the CI check to link to that documentation.